src/mono/mono/mini/jit-icalls.c

```diff
@@ -39,6 +39,18 @@ mono_ldftn (MonoMethod *method)
 	gpointer addr;
 	ERROR_DECL (error);
 
+	if (G_UNLIKELY ((method->flags & METHOD_ATTRIBUTE_PINVOKE_IMPL) == 0 &&
+			method->wrapper_type == MONO_WRAPPER_NONE &&
+			mono_method_has_unmanaged_callers_only_attribute (method))) {
```

Comment:
There's a perf problem here
https://github.com/mono/mono/pull/20060#discussion_r449288732

It's actually worse for the JIT than for the interpreter (interp adds the wrapper at transform time, not at execution time; the JIT does this attribute lookup every time `mono_ldftn` gets called at execution time).

I think we can improve things a bit by checking for the attribute in `method-to-ir` and calling a slightly different icall that unconditionally returns the addr of a wrapper.